### PR TITLE
(MAINT) Extract validate-regex-backreferences!

### PR DESCRIPTION
**Pull request by nwolfe**
This commit extract the regex back-reference & capture group validation
code from validate-auth-config-rule! into its own function.

### CODE REVIEW

**Comment by nwolfe** [comment]
@dankreek t'was bugging me not in it's own function

File: src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj [hunk 1]
```diff
@@ -92,6 +92,42 @@
...
+   to a capture group that doesn't exist."
+  [rule]
+  (let [group-count (try
+                      (-> rule
+                          :match-request
+                          :path
+                          re-pattern
+                          (.matcher "")
+                          .groupCount)
+                      (catch PatternSyntaxException e
```
Comment (nwolfe) [inline src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj:107]: This was originally around the whole function body, but only these first two lines could throw this.
I moved the catch up around only those lines, and then combined the two lines into a `->`.

Happy to reformat this any way, mainly just wanted to reduce the scope of the try/catch

**Comment by camlow325** [comment]
:+1: @dankreek - any concerns?

**Comment by dankreek** [comment]
:+1: I can dig it!

**Pull request merged**